concordium-consensus/src/Concordium/GlobalState/Persistent/Account/ProtocolLevelTokens.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE BinaryLiterals #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Concordium.GlobalState.Persistent.Account.ProtocolLevelTokens where
+
+import qualified Concordium.Crypto.SHA256 as Hash
+import Concordium.GlobalState.Basic.BlockState.LFMBTree (hashAsLFMBTV1)
+import Concordium.GlobalState.Persistent.BlobStore
+import Concordium.GlobalState.Persistent.BlockState.ProtocolLevelTokens
+import Concordium.Types.HashableTo
+import Concordium.Utils.Serialization
+import Control.Monad
+import qualified Data.ByteString as BS
+import qualified Data.Map.Strict as Map
+import Data.Serialize
+
+-- | The table of PLT account states. The table is indexed by the token index
+--  into the global token table.
+newtype TokenAccountStateTable = TokenAccountStateTable
+    { tokenAccountStateTable :: Map.Map TokenIndex (HashedBufferedRef TokenAccountState)
+    }
+    deriving newtype (Show)
+
+instance (MonadBlobStore m) => MHashableTo m Hash.Hash TokenAccountStateTable where
```

Comment:
Should hash to `TokenStateTableHash` instead of `Hash.Hash`.

Reply:
done.